rcl/test/rcl/test_lexer_lookahead.cpp

```diff
@@ -0,0 +1,317 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <string>
+
+#include "../scope_exit.hpp"
+
+#include "rcl/error_handling.h"
+#include "rcl/lexer_lookahead.h"
+
+#ifdef RMW_IMPLEMENTATION
+# define CLASSNAME_(NAME, SUFFIX) NAME ## __ ## SUFFIX
+# define CLASSNAME(NAME, SUFFIX) CLASSNAME_(NAME, SUFFIX)
+#else
+# define CLASSNAME(NAME, SUFFIX) NAME
+#endif
+
+class CLASSNAME (TestLexerLookaheadFixture, RMW_IMPLEMENTATION) : public ::testing::Test
+{
+public:
+  void SetUp()
+  {
+  }
+
+  void TearDown()
+  {
+  }
+};
+
+#define SCOPE_LOOKAHEAD2(name, text) \
+  { \
+    name = rcl_get_zero_initialized_lexer_lookahead2(); \
+    rcl_ret_t ret = rcl_lexer_lookahead2_init(&name, text, rcl_get_default_allocator()); \
+    ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe(); \
+  } \
+  auto __scope_lookahead2_ ## name = make_scope_exit( \
+    [&name]() { \
+      rcl_ret_t ret = rcl_lexer_lookahead2_fini(&buffer); \
+      ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe(); \
+    })
+
+TEST_F(CLASSNAME(TestLexerLookaheadFixture, RMW_IMPLEMENTATION), test_init_fini_twice)
```

Comment:
Based on recent experience, should we add a testcase for calling `rcl_lexer_lookahead2_init` on a non zero-initialized buffer?

Reply:
Added test for non zero-initialized buffer in 69f1f2a4d58b0044b1d2de7dd872290aca7b3fd3